### ISSUE TITLE
The Command.prototype.parse() method has issues.

### ISSUE THREAD

**Issue by yossan** [tj/commander.js#995]
This method has the following issues:

* when registered `subcommand`, thier arguments and options isn't parsed to check valid.
* when registered `action`, `this.parseArgs()` called subcommand without parsing their arguments and options.

As a result, this method's arguments are not evaluted correctly in subcommand.
In the following code expect to occur `error: missing required argument "file"` but not now.

```
parse
  .command('install <file>')
  .parse('node', 'main.js', 'install') // Expect error but not.
```

Also now `the required argument` will not correctly work. 
[Fix to fail if required arguments don't exist.](https://github.com/tj/commander.js/pull/941)

So it seems that this method should be reconsidered fundamentally.

Thank you for the wonderful library.

# Related 

https://github.com/tj/commander.js/pull/941

**Comment by shadowspawn** [comment]
Thanks @yossan. There are some deliberate differences in behaviour between `.action` and separate executable sub-commands, but I think there are also a number of shortcomings and accidental differences. Bring them closer together and detecting more errors is on my wish list.

**Comment by yossan** [comment]
@shadowspawn 

Thank you for the review.
I want "the required argument" to work correctly.
It's a really great library.

**Comment by shadowspawn** [comment]
A missing required argument for a command now shows an error in the published pre-release of Commander v5.0.0 (#1163)

**Comment by shadowspawn** [comment]
Commander v5.0.0 has been released with significant changes to the parsing, and adds default detection of a missing required argument.

https://github.com/tj/commander.js/releases/tag/v5.0.0
